src/application/common/endpoint.ts

```diff
@@ -24,7 +27,21 @@ export class Endpoint {
     }
 
     protected get host() {
-        return location.host || "127.0.0.1:3000"
+
+
+        if (location.host === "") {
+            this.fs.readFile('port', "utf8", function (err: string, data: any) {
```

Comment:
With the latest merged PR port and host are provided to `generator-theia` via the yeoman configuration file: https://github.com/theia-ide/theia/blob/master/examples/browser/.yo-rc.json

We could generate the webpack config and the backend code are using the same port and host for the web with `generator-theia`.

For electron i think it should be possible to open the window with a properly configured location (that contains host) after the express has been started and the port is known.